core/src/com/unciv/logic/map/mapgenerator/MapGenerator.kt

```diff
@@ -596,6 +651,146 @@ class MapGenerator(val ruleset: Ruleset) {
                     -1f, ruleset.modOptions.constants.spawnIceBelowTemperature,
                     0f, 1f))
             }.toList()
+
+        // Flat Earth needs a 1 tile wide perimeter of ice/mountain/snow and a 2 radius cluster of ice in the center.
+        if (tileMap.mapParameters.shape === MapShape.flatEarth) {
```

Comment:
You have 150 lines of uninterrupted code here.
Not only does this need its own function, it probably needs subfunctions, maybe its own class